### PR TITLE
feat(api): add slug parser in the api requests

### PR DESCRIPTION
**Pull request by djabarovgeorge**
### What changed? Why was the change needed?

**The PR is not ready please do not review**

### Screenshots

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Related enterprise PR
[EE-PR](https://github.com/novuhq/packages-enterprise/pull/235)

### Special notes for your reviewer

</details>

### CODE REVIEW

File: apps/api/src/app/workflows-v2/pipes/parse-slug-Id.pipe.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  if (validId) {
+    return validId;
+  }
+
+  const encodedValue = value.slice(-16);
+  let decodedValue: string;
+  try {
+    decodedValue = decodeBase62(encodedValue);
+  } catch (error) {
+    return value;
```
Comment (djabarovgeorge) [inline apps/api/src/app/workflows-v2/pipes/parse-slug-Id.pipe.ts:44]: if not base 62

File: apps/api/package.json [hunk 2]
```diff
@@ -57,6 +57,7 @@
     "@types/newrelic": "^9.14.0",
     "@upstash/ratelimit": "^0.4.4",
     "axios": "^1.6.8",
+    "base-x": "^5.0.0",
```
Comment (rifont) [inline apps/api/package.json]: ⚠ **issue:** Can we write our own base62 encoder from scratch? The pipe logic is effectively using unvalidated data. I'm concerned that any vulnerabilities (intentionally or maliciously) added to the `base-x` package could expose us‏ to an attack vector

File: apps/api/src/app/shared/helpers/base62.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  const uint8Array = decode(encoded);
+
+  return Buffer.from(uint8Array).toString('hex');
+}
+
+export function isBase62(input: string): boolean {
+  const base62Regex = /^[0-9A-Za-z]+$/;
+
+  return base62Regex.test(input);
+}
```
Comment (rifont) [inline apps/api/src/app/shared/helpers/base62.ts]: ⚠ **issue:** Beware, [`base-x` has some logic](https://github.com/cryptocoinjs/base-x/blob/64e196c84f1d0b51eb7650d3881868568a0f525c/ts_src/index.ts#L39) to "skip" leading zeros when encoding/decoding. We should add tests for this case.‏

File: apps/api/package.json (hunk 2 above)
Comment (djabarovgeorge) [reply apps/api/package.json]: What do you think if will have fixed version 5.0.0? Will it solve the concern for us? Then, we could upgrade the version if needed and only to a stable and secure version.

Comment (SokratisVidros) [reply apps/api/package.json]: Is there a know security issue with the package?

Comment (SokratisVidros) [reply apps/api/package.json]: The truth is that a custom implementation with two functions, encodeBase62, decodeBase62 would be much much simpler, no need for pipes, no need to worry about potential issues.

**Review by SokratisVidros — CHANGES_REQUESTED** [review]
I am requesting changes only because of base-x

File: apps/api/package.json (hunk 2 above)
Comment (djabarovgeorge) [reply apps/api/package.json]: > ⚠ issue: Can we write our own base62 encoder from scratch? The pipe logic is effectively using unvalidated data. I'm concerned that any vulnerabilities (intentionally or maliciously) added to the base-x package could expose us‏ to an attack vector

i had a hard time implementing the code from scratch, i added up comping the code from base-x so we won't get any vulnerabilities by mistake.

> Is there a know security issue with the package?

not at the moment.


> The truth is that a custom implementation with two functions, encodeBase62, decodeBase62 would be much much simpler, no need for pipes, no need to worry about potential issues.

i implemented encodeBase62, decodeBase62 based on base-x, pipes are actually a nice touch of nest js making the boundaries of controller and use case responsibility cleaner

File: apps/api/src/app/shared/helpers/base62.ts (hunk 3 above)
Comment (djabarovgeorge) [reply apps/api/src/app/shared/helpers/base62.ts]: mongo object is the first byte of the timestamp component, and the timestamp component is always a positive integer. however in order to make sure i added tests, from the tests i found out that the encoding supports 2 leading zero.

**Review by SokratisVidros — APPROVED** [review]

**Pull request merged**